### PR TITLE
Add articles route and tests

### PR DESCRIPTION
**Pull request by kamalsacranie**
Added the /api/articles endpoint with happy and sad path testing

### CODE REVIEW

File: __tests__/articlesRoutes.tests.js [hunk 1]
```diff
@@ -47,3 +47,54 @@ describe("GET /api/articles/:article_id", () => {
     });
   });
 });
+
+describe("GET /api/articles", () => {
+  describe("Happy path", () => {
+    it("should return a 200", async () => {
+      await request(app).get("/api/articles").expect(200);
+    });
+    it("should return an array of objects", async () => {
```
Comment (roisinoreilly) [inline __tests__/articlesRoutes.tests.js]: is that what you're testing here? You've just tested to see if it's an array with length greater than 0

File: __tests__/articlesRoutes.tests.js [hunk 2]
```diff
@@ -47,3 +47,54 @@ describe("GET /api/articles/:article_id", () => {
...
+        body: { articles },
+      } = await request(app).get("/api/articles").expect(200);
+      expect(articles).toBeInstanceOf(Array);
+      expect(articles).not.toHaveLength(0);
+    });
+    it("should return a list of objects that match the article schema", async () => {
+      const {
+        body: { articles },
+      } = await request(app).get("/api/articles").expect(200);
+      articles.forEach((article) => {
```
Comment (roisinoreilly) [inline __tests__/articlesRoutes.tests.js:67]: I know you've checked the length in a previous assertion, but any time you're iterating in a test I would stack the expect statements to check the length then iterate in the same test block

File: __tests__/articlesRoutes.tests.js [hunk 3]
```diff
@@ -47,3 +47,54 @@ describe("GET /api/articles/:article_id", () => {
...
+        expect(article).toMatchObject({
+          author: expect.any(String),
+          title: expect.any(String),
+          article_id: expect.any(Number),
+          body: expect.any(String),
+          topic: expect.any(String),
+          created_at: expect.any(String || undefined),
+          votes: expect.any(Number),
+          article_img_url: expect.any(String || undefined),
+          comment_count: expect.any(String),
```
Comment (roisinoreilly) [inline __tests__/articlesRoutes.tests.js:77]: great job here, I'd also like to check that the articles are sorted by descending date (jest sorted is great)

File: src/routes/articles.js [hunk 4]
```diff
@@ -1,8 +1,12 @@
 const express = require("express");
-const { getArticle } = require("../controllers/articles");
+const { getArticle, getArticles } = require("../controllers/articles");
+const { sqlError } = require("./errors");
 
 const articlesRouter = express.Router();
```
Comment (roisinoreilly) [inline src/routes/articles.js:5]: very impressive

File: src/models/articles.js [hunk 5]
```diff
@@ -8,3 +8,18 @@ exports.fetchArticle = async (article_id) => {
   );
   return responseRowsOr404(query, "article not found");
 };
+
+exports.fetchArticles = async () => {
+  const query = await db.query(
+    `
+      SELECT articles.*, COUNT(comments.article_id) as comment_count FROM articles
```
Comment (roisinoreilly) [inline src/models/articles.js:15]: nicely handled

File: __tests__/articlesRoutes.tests.js [hunk 6]
```diff
@@ -47,3 +47,54 @@ describe("GET /api/articles/:article_id", () => {
...
+        body: { articles },
+      } = await request(app).get("/api/articles").expect(200);
+      articles.forEach((article) => {
+        expect(article).toMatchObject({
+          author: expect.any(String),
+          title: expect.any(String),
+          article_id: expect.any(Number),
+          body: expect.any(String),
+          topic: expect.any(String),
+          created_at: expect.any(String || undefined),
```
Comment (roisinoreilly) [inline __tests__/articlesRoutes.tests.js:74]: I think you could use expect.toBeDate() here to make your test more robust

File: __tests__/articlesRoutes.tests.js (hunk 3 above)
Comment (roisinoreilly) [inline __tests__/articlesRoutes.tests.js:77]: I would also expect this to be a number

**Review by roisinoreilly — APPROVED** [review]
really great work! I've left a few comments for changes you might want to implement, mainly just testing that the articles are being sorted by date descending. After you've done that feel free to merge :)

File: __tests__/articlesRoutes.tests.js (hunk 1 above)
Comment (kamalsacranie) [reply __tests__/articlesRoutes.tests.js]: Agreed. Will update description

**Pull request merged**